lib/fastlane/plugin/wpmreleasetoolkit/helper/android/android_version_helper.rb

```diff
@@ -30,42 +30,54 @@ module VersionHelper
         #         - If this version is a hotfix (more than 2 parts and 3rd part is non-zero), returns the "X.Y.Z" formatted string
         #         - Otherwise (not a hotfix / 3rd part of version is 0), returns "X.Y" formatted version number
         #
-        def self.get_public_version
-          version = get_release_version
+        def self.get_public_version(app)
+          version = get_version_from_properties(app, false)
           vp = get_version_parts(version[VERSION_NAME])
           return "#{vp[MAJOR_NUMBER]}.#{vp[MINOR_NUMBER]}" unless is_hotfix?(version)
 
           "#{vp[MAJOR_NUMBER]}.#{vp[MINOR_NUMBER]}.#{vp[HOTFIX_NUMBER]}"
         end
 
-        # Extract the version name and code from the `vanilla` flavor of the `$PROJECT_NAME/build.gradle file`
-        #   or for the defaultConfig if `HAS_ALPHA_VERSION` is not defined.
+        # Extract the version name and code from the release version of the app from `version.properties file`
         #
-        # @env HAS_ALPHA_VERSION If set (with any value), indicates that the project uses `vanilla` flavor.
+        # @return [Hash] A hash with 2 keys "name" and "code" containing the extracted version name and code, respectively
+        #
+        def self.get_release_version(app)
+          return get_version_from_properties(app, false)
+        end
+
+        # Extract the version name and code from the `version.properties` file in the project root
         #
         # @return [Hash] A hash with 2 keys "name" and "code" containing the extracted version name and code, respectively
         #
-        def self.get_release_version
-          section = ENV['HAS_ALPHA_VERSION'].nil? ? 'defaultConfig' : 'vanilla {'
-          gradle_path = self.gradle_path
-          name = get_version_name_from_gradle_file(gradle_path, section)
-          code = get_version_build_from_gradle_file(gradle_path, section)
-          return { VERSION_NAME => name, VERSION_CODE => code }
+        def self.get_version_from_properties(product_name, is_alpha)
```

Comment:
nit: When reading the call sites, it was really unclear to me what the 2nd parameter was supposed to be, until I looked at the implementation line here. What do you think about using named parameters for that method instead, so that the call site end up being more readable? Would also allow you to provide a default value for `is_alpha` as a bonus.

```suggestion
        def self.get_version_from_properties(product_name:, is_alpha: false)
```